feature/system/gnmi/set/tests/gnmi_set_test/gnmi_set_test.go

```diff
@@ -157,14 +190,6 @@ func TestDeleteInterface(t *testing.T) {
 		config.DeleteInterface(p1.Name())
 		config.DeleteInterface(p2.Name())
 
-		for _, iname := range scope.interfaces {
-			iface := config.GetInterface(iname)
-			if iface == nil {
-				config.Interface = nil
-
-			}
-		}
-
 		op.push(t, dut, config, scope)
```

Comment:
This part of code was never present in the initial version of the tc and the adding it breaks us. I think if any vendor needs it, it should be part of their deviation.

Reply:
As per gnmi specification guidelines replace function can not have empty jason object.
Both at root level and container level op.push() function calls replace function and it becomes invalid.

It is true that above code was not in original script ,  we have added as per the specification guidelines mentioned.

Reply:
Agreed, function cannot have empty json objects, but that clause is exactly doing the opposite and passing empty interfaces which eventually fails at root/container interface level.

Reply:
Yes , please add this code back since it is as per Specification.
You need to add deviation here rather than deleting the code and Software needs to provide fix as it is clearly mentioned not to accept empty Jason object and through error as "invalid object".

Reply:
Apologies, if I was not clear. I meant that code leads to passing empty JSON obj and hence it fails for us also it is diverting from the original version.
 
Probably that code is needed for a specific vendor software in which case it is their deviation, by default we don't need that code (like the initial version) since we are not passing empty data.

Reply:
If you debug the see at this line , it is setting empty jason object [] to interfaces to delete container which is wrong as per specification. 
 "nil" is approved in specification to empty objects and not {} / [].

Reply:
By default when code is absent we have interface data present under config map and not empty, indeed if any vendor needs to set it to nil it is their deviation.

Reply:
Please check this as in specficiation doc.

replace MUST not be used as a way to delete configuration at the path specified by being supplied with a null or invalid value. For example, it is invalid to replace a keyed list element (e.g. /a/f[k=10]) with an empty JSON object {}, which implicitly deletes the list's keys and renders the list element keyless.

Reply:
@Ankur19  Please could you check this part of code. We have discussed  with you and got your approval before adding it when we merged it via pull request #2328.  
This is as per specification mentioned in gnmi. 

replace MUST not be used as a way to delete configuration at the path specified by being supplied with a null or invalid value. For example, it is invalid to replace a keyed list element (e.g. /a/f[k=10]) with an empty JSON object {}, which implicitly deletes the list's keys and renders the list element keyless.

Reply:
Here is the value of config.Interface.

Before:

*oc.Root {Acl: *github.com/openconfig/ondatra/gnmi/oc.Acl nil, Component: map[string]*github.com/openconfig/ondatra/gnmi/oc.Component nil, DefinedSets: *github.com/openconfig/ondatra/gnmi/oc.DefinedSets nil, Flow: map[string]*github.com/openconfig/ondatra/...
oc.Root {Acl: *github.com/openconfig/ondatra/gnmi/oc.Acl nil, Component: map[string]*github.com/openconfig/ondatra/gnmi/oc.Component nil, DefinedSets: *github.com/openconfig/ondatra/gnmi/oc.DefinedSets nil, Flow: map[string]*github.com/openconfig/ondatra/g...
Acl: *oc.Acl nil
Component: map[string]*oc.Component nil
DefinedSets: *oc.DefinedSets nil
Flow: map[string]*oc.Flow nil
Interface: map[string]*oc.Interface []      >>>>>>>>>>>>>>>>>>   Before replacing.
Keychain: map[string]*oc.Keychain nil
Lacp: *oc.Lacp nil
Lldp: *oc.Lldp nil
Meta: *oc.Meta nil
NetworkInstance: map[string]*oc.NetworkInstance ["DEFAULT": *(*"github.com/openconfig/ondatra/gnmi/oc.NetworkInstance")(0xc002a18f20), ]
Qos: *oc.Qos nil
RoutingPolicy: *oc.RoutingPolicy nil
Sampling: *oc.Sampling nil
System: *oc.System nil
TerminalDevice: *oc.TerminalDevice nil


After:

config
*oc.Root {Acl: *github.com/openconfig/ondatra/gnmi/oc.Acl nil, Component: map[string]*github.com/openconfig/ondatra/gnmi/oc.Component nil, DefinedSets: *github.com/openconfig/ondatra/gnmi/oc.DefinedSets nil, Flow: map[string]*github.com/openconfig/ondatra/...
oc.Root {Acl: *github.com/openconfig/ondatra/gnmi/oc.Acl nil, Component: map[string]*github.com/openconfig/ondatra/gnmi/oc.Component nil, DefinedSets: *github.com/openconfig/ondatra/gnmi/oc.DefinedSets nil, Flow: map[string]*github.com/openconfig/ondatra/g...
Acl: *oc.Acl nil
Component: map[string]*oc.Component nil
DefinedSets: *oc.DefinedSets nil
Flow: map[string]*oc.Flow nil
Interface: map[string]*oc.Interface nil                      >>>>>>>>>>> After replacing.
Keychain: map[string]*oc.Keychain nil
Lacp: *oc.Lacp nil
Lldp: *oc.Lldp nil
Meta: *oc.Meta nil
NetworkInstance: map[string]*oc.NetworkInstance ["DEFAULT": *(*"github.com/openconfig/ondatra/gnmi/oc.NetworkInstance")(0xc002a18f20), ]
Qos: *oc.Qos nil
RoutingPolicy: *oc.RoutingPolicy nil
Sampling: *oc.Sampling nil
System: *oc.System nil
TerminalDevice: *oc.TerminalDevice nil

Reply:
If you see above other elements in config , if there is no value it is set to nil not [].

Reply:
I would agree with @manan-patel that the removed code is incorrect

The loop/nil check that you used to nullify the interfaces config doesn't make much sense, and it removes config for all interfaces (if there are any other configured interfaces present in addition to the two that were removed). This will break, for instance, mgmt configuration if your implementation exposes mgmt interfaces in OC.

What you can do instead is
```
if len(config.Interface) == 0 {
    config.Interface = nil
}
```

This makes much more sense, and I think it shouldn't break implementations like ours and @manan-patel that have other interfaces exposed in OC. 

I'll leave it up to Google to decide if this should be a deviation or not,

Reply:
Thanks @LimeHat , I did not think of vendors who have management interfaces exposed in OC.

The suggestion you have provided definitely works for us . 
@manan-patel , please could you update this code which @LimeHat  suggested? 

if len(config.Interface) == 0 {
    config.Interface = nil
}

Reply:
It works for us, please add the deviation if google suggests for it later.

Reply:
Thanks @manan-patel.